### PR TITLE
fix: close file in error handling

### PR DESCRIPTION
**Pull request by testwill**

### CODE REVIEW

**Review by machine424 — APPROVED** [review]
Thanks!

File: promql/query_logger.go [hunk 1]
```diff
@@ -96,12 +96,14 @@ func getMMapedFile(filename string, filesize int, logger log.Logger) ([]byte, io
 
 	err = file.Truncate(int64(filesize))
 	if err != nil {
+		file.Close()
```
Comment (SuperQ) [inline promql/query_logger.go:99]: Why not `defer file.Close()`?

Comment (aknuds1) [reply promql/query_logger.go:99]: @SuperQ `file` is only closed in error cases in this function, why `defer` as you don't want to close the file if all went well?

File: promql/query_logger_test.go [hunk 2]
```diff
@@ -110,10 +110,7 @@ func TestMMapFile(t *testing.T) {
 	filename := file.Name()
 	defer os.Remove(filename)
 
-	fileAsBytes, closer, err := getMMapedFile(filename, 2, nil)
-	if err != nil {
-		t.Cleanup(func() { closer.Close() })
-	}
```
Comment (aknuds1) [inline promql/query_logger_test.go:116]: What I don't understand here is, why not close `closer` at the end of the test? If there's no error, there will be a valid `closer`, right?

**Review by aknuds1 — COMMENTED** [review]
I don't understand why not close the returned `Closer` in `TestMMapFile`.

File: promql/query_logger.go (hunk 1 above)
Comment (machine424) [reply promql/query_logger.go:99]: https://man7.org/linux/man-pages/man2/mmap.2.html
```
After the mmap() call has returned, the file descriptor, fd, can
       be closed immediately without invalidating the mapping.
```

Comment (machine424) [reply promql/query_logger.go:99]: yes that could be tested/done in another PR.

Comment (aknuds1) [reply promql/query_logger.go:99]: Agree not to mix topics in the PR :)

File: promql/query_logger_test.go (hunk 2 above)
Comment (machine424) [reply promql/query_logger_test.go:116]: Good catch, I bet `defer os.Remove(filename)` in 111 is already returning an error sometimes (on windows).

Comment (aknuds1) [reply promql/query_logger_test.go:116]: I also don't understand FWIW why `file` is not closed immediately, as you are just using the name.

Comment (machine424) [reply promql/query_logger_test.go:116]: his also seems unrelated to the PR. We can address it in that follow-up PR.

File: promql/query_logger.go (hunk 1 above)
Comment (machine424) [reply promql/query_logger.go:99]: Could this get another approval then?

File: promql/query_logger_test.go (hunk 2 above)
Comment (aknuds1) [reply promql/query_logger_test.go:116]: @machine424 I think this should be changed into 
```gol
	fileAsBytes, closer, err := getMMapedFile(filename, 2, nil)
	if err == nil {
		t.Cleanup(func() { closer.Close() })
	}
```

File: promql/query_logger.go (hunk 1 above)
Comment (aknuds1) [reply promql/query_logger.go:99]: Approved.

**Review by SuperQ — APPROVED** [review]

**Pull request merged**